pkg/client/dtclient/settings_client.go

```diff
@@ -1052,9 +1049,8 @@ func (d *SettingsClient) DeletePermission(ctx context.Context, objectID string)
 		corerest.RequestOptions{CustomShouldRetryFunc: corerest.RetryIfTooManyRequests},
 	))
 
-	apiError := coreapi.APIError{}
 	// deployments with "none" for all-user will always try to delete. This could be an update (restricted to shared) or it stays the same (delete 404)
-	if err != nil && (!errors.As(err, &apiError) || apiError.StatusCode != http.StatusNotFound) {
+	if err != nil && !coreapi.IsNotFoundError(err) {
```

Comment:
First part of the if is not needed, as if there is a 404 err the error is also always not nil.

Reply:
Here it's actually needed because we are checking for "not" 404.
`IsNotFoundError` returns false if the error is nil, meaning that `!coreapi.IsNotFoundError(err)` will resolve to true if the error is nil

Reply:
`coreapi.IsNotFoundError(err)` is basically `err != nil && coreapi.IsNotFoundError(err)`
whereas
`!coreapi.IsNotFoundError(err)` is basically `err == nill || !coreapi.IsNotFoundError(err)`